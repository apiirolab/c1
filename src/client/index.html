<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>