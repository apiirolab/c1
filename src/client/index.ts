export const info = 'Client Info';
