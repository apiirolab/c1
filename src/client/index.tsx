import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
