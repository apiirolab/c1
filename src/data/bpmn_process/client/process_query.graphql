query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
