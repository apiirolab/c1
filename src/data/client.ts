import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
