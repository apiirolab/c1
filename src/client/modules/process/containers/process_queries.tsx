import * as PROCESSES_QUERY from 'data/bpmn_process/client/processes_query.graphql';

import { Query, QueryProps } from 'react-apollo';

import { Yoga } from 'data/yoga';

export { PROCESSES_QUERY };

/* =========================================================
    Notifications
   ======================================================== */

type Data = { processes: Yoga.BpmnProcess[] };
type Variables = { input: Yoga.BpmnProcessesInput };
type Props = Partial<QueryProps<Data, Variables>>;

export class ProcessListQuery extends Query<Data, Variables> {
  static displayName = 'ProcessesQuery';
  static defaultProps: Props = {
    query: PROCESSES_QUERY,
    variables: {
      input: {
        first: 50
      }
    }
  };
}
