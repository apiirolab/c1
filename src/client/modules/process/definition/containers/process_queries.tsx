import * as PROCESS_LIST_QUERY from '../queries/bpmn_processes_query.graphql';

import { Query, QueryProps } from 'react-apollo';

import { QueryTypes } from 'data/client';

export { PROCESS_LIST_QUERY };

/* =========================================================
    Process List
   ======================================================== */

type Data = QueryTypes.ProcessesQuery;
type Variables = QueryTypes.ProcessesQueryVariables;
type Props = Partial<QueryProps<QueryTypes.ProcessesQuery, QueryTypes.ProcessesQueryVariables>>;

export class ProcessListQuery extends Query<Data, Variables> {
  static displayName = 'ProcessesQuery';
  static defaultProps: Props = {
    query: PROCESS_LIST_QUERY,
    variables: {
      input: {
        first: 50
      }
    }
  };
}
