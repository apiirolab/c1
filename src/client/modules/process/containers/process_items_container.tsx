import * as React from 'react';

import { inject } from 'mobx-react';

import { renderResult } from 'client/modules/common';
import { ProcessListView } from '../views/process_items_view';
import { ProcessListQuery } from './process_queries';

type Props = {
  path?: string;
  context?: App.Context;
};

export const ProcessListContainer: React.SFC<Props> = inject('context')(({ context }) => (
  <ProcessListQuery>
    {result => {
      // console.log(result.data.processes)
      return renderResult(result, () => (
        <ProcessListView processes={result.data.processes} context={context} />
      ));
    }}
  </ProcessListQuery>
));

ProcessListContainer.displayName = 'ProcessListContainer';
