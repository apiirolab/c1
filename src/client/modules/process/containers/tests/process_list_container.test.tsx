import * as React from 'react';

import { mock, MockedProvider, render } from 'client/tests';

import { ProcessListContainer } from '../process_items_container';
import { PROCESS_LIST_QUERY } from '../process_queries';
import { createProcesses } from './process_test_data';

describe('Process', () => {
  describe('List', () => {
    describe('Container', () => {
      beforeEach(() => mock.reset());

      function componentWithData() {
        return (
          <MockedProvider>
            <div>
              <ProcessListContainer />
            </div>
          </MockedProvider>
        );
      }

      function luisComponent() {
        mock.expect(PROCESS_LIST_QUERY).reply({
          processes: createProcesses()
        });

        return componentWithData();
      }

      it('renders loading', () => {
        mock.expect(PROCESS_LIST_QUERY).loading();
        const root = render(componentWithData());
        expect(root).toMatchSnapshot();
      });

      it('renders error', () => {
        mock.expect(PROCESS_LIST_QUERY).fail('This is some error');
        const root = render(componentWithData());
        expect(root).toMatchSnapshot();
      });

      it('renders data', async () => {
        mock.expect(PROCESS_LIST_QUERY).reply({
          processes: createProcesses()
        });

        const root = render(componentWithData());
        expect(root).toMatchSnapshot();
      });

      return {
        component: luisComponent()
      };
    });
  });
});
