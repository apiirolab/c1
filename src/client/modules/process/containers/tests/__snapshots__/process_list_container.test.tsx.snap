// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process List Container renders data 1`] = `
Array [
  <div
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="find icon"
    />
    <div
      className="content"
    >
      Search
      <div
        className="sub header"
      >
        Find process resources
      </div>
    </div>
  </div>,
  <div
    className="ui divided relaxed list sc-bdVaJa bviDoP"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="file outline icon middle aligned"
      />
      <div
        className="content"
      >
        <a
          className="header"
        />
        <a
          className="description"
        >
          Submitted by 
           路 
           路 
          Thu, 01 Feb 2018 23:00:00 GMT
        </a>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <i
        aria-hidden="true"
        className="file outline icon middle aligned"
      />
      <div
        className="content"
      >
        <a
          className="header"
        />
        <a
          className="description"
        >
          Submitted by 
           路 
           路 
          Thu, 01 Feb 2018 23:00:00 GMT
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`Process List Container renders loading 1`] = `
<p>
  Loading...
</p>
`;
